source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,352 @@
+entity
+* Security group rules: specify the access to a network resource;
+networking API entity
+
+### Benefit to oVirt
+
+Currently, all incoming or outgoing traffic to oVirt VMs is allowed,
+which - at best - is a security risk. This lax practice could lead to
+security breaches and information theft in the oVirt clusters.
+
+Security groups, which are a traffic white list, help protect against
+these issues by specifying which traffic is allowed to *and* from the
+resources.
+
+## Objectives
+
+* Provide a backwards compatible security group implementation, using OVN
+as the backend. If possible, no additional backend / database should be
+used.
+
+* When possible, Openstack neutron's behavior should be mimicked: by
+default, all VM incoming IP traffic is dropped, and all outgoing IP
+traffic is allowed.
+
+* The security group feature should be configurable, and activated by
+default - this means an option for using oVirt's external networks
+without security group support should be possible.
+
+* If the external network has a subnet defined, DHCP traffic is allowed to
+/ from the VM.
+
+### Requirements
+
+* Add a security group
+* List all security groups
+* Show a particular security group
+* Update a security group
+* Remove a security group
+* Apply a security group to a (set of) port(s)
+* Add a security group rule to a security group
+* Delete security group rule(s) from a security group
+* List security group rules
+* Show a particular security group rule
+* Deny all incoming VM traffic ***if*** configured accordingly
+* Allow all DHCP related traffic ***if*** there's a subnet associated to
+that port's network
+
+### Security groups description
+
+Security groups, in the general sense, are nothing more than containers
+for security group rules. They are applied to logical ports, and thus
+are used to associate a set of rules to a port.
+
+They are are indeed a white list - e.g. everything is *denied* by
+default, and the user adds rules that allow traffic. As long as *one*
+rule is matched, the traffic is allowed.
+
+Security group rules can match on L3 and L4 parameters, specifically on
+the protocol, the IP protocol, and destination ports. The remote IP
+address is also a subject for the match conditions, and rules can be
+added to allow *incoming* and *outgoing* traffic.
+
+More information on the API of security group rules can be found
+[here](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules).
+
+### Activating the feature
+
+Two alternatives are possible:
+
+1. Add a configuration flag to the ovirt-provider-ovn configuration file.
+
+2. Use the [Allowed address pairs](https://developer.openstack.org/api-ref/network/v2/#allowed-address-pairs) networking API extension.
+Using this extension, we would be able to individually specify to which ports we would want to activate the 'deny-all' behavior, instead activating it for all ports.
```

Comment:
There is a third option: to add allow-all rules to existing networks during the upgrade.